### PR TITLE
use jest directly instead of dts-cli

### PR DESCRIPTION
**Pull request by zxbodya**
Updating project configurations to use jest directly instead of dts-cli

### Reasons for making this change

This allows better jest IDE integrations to work out of the box.

Also updating to current jest version.

### CODE REVIEW

File: packages/validator-ajv8/jest.config.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ module.exports = {
   coveragePathIgnorePatterns: ['<rootDir>/node_modules/', '<rootDir>/test'],
   coverageThreshold: {
     global: {
-      branches: 100,
+      // todo: dropped from 100 after jest 29
+      branches: 98,
```
Comment (zxbodya) [inline packages/validator-ajv8/jest.config.js:15]: there is improvement in coverage reporting in new jest causing drop in branch coverage

**Review by nickgros — APPROVED** [review]

File: packages/validator-ajv8/jest.config.js (hunk 1 above)
Comment (heath-freenome) [reply packages/validator-ajv8/jest.config.js:15]: Please file a new issue to restore this back to 100%

**Comment by heath-freenome** [comment]
@zxbodya I'm not sure what's going on but your tests seem to be running without completion for both node 16 and 18. Is that the case locally? I'm guessing not. Can you investigate?

**Comment by zxbodya** [comment]
yes, it works locally. will look into it

**Comment by zxbodya** [comment]
found the issue:

```
@rjsf/core: Jest did not exit one second after the test run has completed.
@rjsf/core: 
@rjsf/core: 'This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

this was caused by something in jsdom environment setup, switching to use `testEnvironment: 'jsdom'` in jest config - fixed it

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**